README.md

```diff
@@ -1,2 +1,156 @@
-# action
-GitHub Action for code bases using Git Town's stashed commits feature
+<p align="center">
+  <picture>
+    <source media="(prefers-color-scheme: light)" srcset="https://raw.githubusercontent.com/git-town/git-town/main/website/src/logo.svg">
+    <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/git-town/git-town/main/website/src/logo-dark.svg">
+    <img alt="Git Town logo" src="https://raw.githubusercontent.com/git-town/git-town/main/website/src/logo.svg">
+  </picture>
+</p>
+
+# Git Town Action V1
+
+This action synchronizes visualizes your stacked changes when proposing pull requests on GitHub. This allows you to easily see all related PRs for a given pull request, where
+you are in the stack, as well as navigate between PRs in a stack.
+
+It is designed to work out of the box with Git Town v12+, but also supports previous versions via the 
+`perennial-branches` input.
+
+# What's New
+
+Please refer to the [release page](https://github.com/actions/checkout/releases/latest) for the latest release notes.
+
+# Usage
+
+<!-- start usage -->
+```yaml
+- uses: git-town/action@v1
+  with:
+    # A GitHub access token, to be provided by either ${{ github.token }} or 
+    # ${{ secrets.GITHUB_TOKEN }}.
+    # Used to fetch your repository's pull requests via Octokit.
+    # 
+    # Default: ${{ github.token }}
+    github-token: ''
+
+    # The perennial branches for your repository.
+    #
+    # Only necessary when not using a shared Git Town config file, as this
+    # extension will read the repository's perennial branches from config
+    # when possible. 
+    perennial-branches: |
+
+    # The regex used by the action to find and replace the visualization
+    # within a PR's body.
+    # 
+    # Default: '[ \t]*?## Stack[\s\S]*?(?=##|$)'
```

Comment:
Make the number of `#` flexible here?

Reply:
Ah yes that didn't even cross my mind!

I'll have to circle back to this last though, as I gave it a crack and couldn't get the lookahead for the next section (`(?=##|$)`) to be flexible as well.

Reply:
I agree this can get tricky. Considering that the next section might not use the same heading level (amount of `#`) as this section, it might be enough to just match for  `^#+`, i.e. any amount of `#` at the beginning of a line.